src/Features/Core/Portable/MoveDeclarationNearReference/AbstractMoveDeclarationNearReferenceService.State.cs

```diff
@@ -33,10 +33,11 @@ private class State
                 TService service,
                 Document document,
                 TLocalDeclarationStatementSyntax statement,
+                bool canMovePastOtherDeclarationStatements,
```

Comment:
out of curiosity, why?  the behavior was about what to do with a declaration that was already in a block of declarations.  Specifically, to not move it since there's no 'better' location in a declaration block, and users may likely order as they see fit.  What's the use case where we want to do this?

Reply:
The existing behavior makes sense for the refactoring, for example you don't want to offer this refactoring for declaration for 'x' in below piece of code as refactoring will continually be offered for 'x' and 'y':
```
  int x = 1;
  int y = 2;
  var z = x + y;
```

However, for the other clients (like the code fix being added in this PR), it would make sense to move the declaration closer skipping other decl statements. For example, consider:
```
  int x = M();
  int y = 2;
  x = 3;
```

Here, the code fix wants to remove the unused value assignment to 'x' in first statement, so replaces it with below before invoking the service:
```
  int x;
  _ = M();
  int y = 2;
  x = 3;
```

We would like the new declaration for 'x' to be moved to the last statement, ignoring presence of intermediate declaration statements, which will not happen unless this flag is true:
```
  _ = M();
  int y = 2;
  int x = 3;
```

I also have unit tests to verify this code path.

---
In reply to: [230603926](https://github.com/dotnet/roslyn/pull/30777#discussion_r230603926) [](ancestors = 230603926)

Reply:
This is a somewhat different case though.  This is just "merge a split decl+assignment into one local-statement". 

1. in that case, i wouldn't have an issue with that.  but that doesn't need a boolean, it just needs to check if the move is for that purpose.
2. in your code examples, your declarations are not next to each other. so i don't see how the new oolean applies.  i.e.:

```c#
  int x;
  _ = M();
  int y = 2;
  x = 3;
```

`int x;` and `int y = 2;` aren't next to each other.  So i don't see why move-decl would *not* be working here in its current state.  It will see that it's not in a decl group and should be ok moving things.  I just tested and that's the behavior i see right now:

![image](https://user-images.githubusercontent.com/4564579/48016923-c7d9c000-e0e1-11e8-9da7-2660f877e36d.png)

Reply:
```
  int x;
  int y = 2;
  x = 3;
```

Please just update MoveDecl to handle this case without a param :)   This should just work.  I just didn't account for this in my cases.  As you point out there is code to block moving things if you have:

```
int a = 1;
int b = 2;
Log(a + b);
```

but this isn't htat case :)

Reply:
Done.

---
In reply to: [230869685](https://github.com/dotnet/roslyn/pull/30777#discussion_r230869685) [](ancestors = 230869685)